operators/advanced_network/advanced_network/managers/gpunetio/adv_network_doca_mgr_obj.cpp

```diff
@@ -274,6 +300,34 @@ doca_error_t DocaRxQueue::create_udp_pipe(const FlowConfig& cfg,
   return DOCA_SUCCESS;
 }
 
+doca_error_t DocaRxQueue::create_rx_packet_list() {
+  doca_error_t result;
+
+  printf("Alloc packet list for Rxq\n");
+
+  result = doca_gpu_mem_alloc(gdev, MAX_DEFAULT_SEM_X_QUEUE * sizeof(struct adv_doca_rx_gpu_info), GPU_PAGE_SIZE, DOCA_GPU_MEM_TYPE_CPU_GPU, (void**)&pkt_list_gpu, (void**)&pkt_list_cpu);
+  if (result != DOCA_SUCCESS) {
+    HOLOSCAN_LOG_ERROR("Failed allocate packet list memory: {}", doca_error_get_descr(result));
+    return DOCA_ERROR_BAD_STATE;
+  }
+  
+  return result;
+}
+
+doca_error_t DocaRxQueue::destroy_rx_packet_list() {
+  doca_error_t result;
+
+  result = doca_gpu_mem_free(gdev, pkt_list_gpu);
+  if (result != DOCA_SUCCESS) {
+    HOLOSCAN_LOG_ERROR("Failed destroy packet list memory: {}", doca_error_get_descr(result));
+    return DOCA_ERROR_BAD_STATE;
+  }
+
+  return result;
+}
+
+
+#if 0
```

Comment:
It seems that by removing the semaphores there's no longer synchronization with the CPU, so now the example assumes the CPU has no idea when packets are received and when to launch kernels?